specs/experimental/plasma.md

```diff
@@ -67,6 +67,16 @@ The [batching][batcher] and compression of input data remain unchanged. When a b
 to be submitted to the inbox address, the data is uploaded to the DA storage layer instead, and a
 commitment (keccak256 hash) is submitted as the bacher inbox transaction call data.
 
+Commitments are encoded as `commitment_type_byte ++ commitment_bytes`, where `commitment_bytes` depends
+on the `commitment_type_byte` where [0, 128] are reserved for official implementations:
+
+| `commitment_type` | `commitment`                    |
+| ----------------- | ------------------------------- |
+| 0                 | `keccak256(tx_payload)`         |
+
+The batcher SHOULD cap input payloads to the maximum L1 calldata size or risk the input to be skipped
```

Comment:
What is the maximum calldata size? There is a policy level for p2p gossip at max tx size but there is no max L1 calldata size besides buying 30 million gas with calldata. What do you mean?

Reply:
I see that you added a line about the max size. It should be called something like max batch size since that is what it describes right? The commitments could be posted in blobs right? So we aren't necessarily tied to calldata

Reply:
I think the default value for `max-l1-tx-size-bytes` makes sense altho it would make it into a consensus parameter rather than being a config option. Perhaps we want to add something to the derivation pipeline and the batcher that checks if its in plasma mode, this adds additional branching logic which isn't ideal but i don't know if we want to add this consensus restriction to rollup mode. The batcher should check the size before the call to `PlasmaDA.SetInput` and error if its larger than that and then maybe on start check the configured param isn't larger than the default  (now consensus) param. Would also need to add something to the derivation pipeline on the get call to check the size of the returned data and throw it out if its too large. This is a slightly annoying edge case to have to test for